src/move_basic.cpp

```diff
@@ -844,16 +867,35 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
             velocity = 0;
         }
 
-        if (abs(remaining.x()) < linearTolerance) {
-            velocity = 0;
-            done = true;
-            ROS_INFO("MoveBasic: Done linear, error %f, %f meters",
-                     remaining.x(), remaining.y());
-        }
+	// Abort checks
+	if ((distRemaining < prevDistance) && (abortTimeout != 0)) {
+		double current = ros::Time::now().toSec();
+		if (current-last > abortTimeout) {
+			abortGoal("MoveBasic: No progress towards goal for longer than timeout.");		
+			sendCmd(0, 0);
+			return false;
+		}
+		else {
+			prevDistance = distRemaining;
+		}
+	}
+	/*
+	if ((distRemaining < prevDistance) && (distThreshold != 0)) {
```

Comment:
Lets try not to leave in commented out code if we don't have to.

Reply:
I can comment it out, but than it would be useless. This is the second method of checking for Goal abort(The first is just above it). We agreed once the two will be tested, one will be chosen.